packages/invoices/src/OutboundShipment/DetailView/reducer.test.ts

```diff
@@ -293,68 +341,73 @@ describe('DetailView reducer', () => {
-
-    // Try find the lines we just deleted
-    const line = state.draft.lines.find(({ id }) => lineToUpdate.id === id);
-    expect(line?.numberOfPacks).toBe(999);
+    expect(line).toEqual(expect.objectContaining({ isDeleted: true }));
   });
 
-  it('adds flags to a line when inserted for being created', () => {
-    const lineToInsert = createLine('999');
-    const state = callReducer(OutboundAction.upsertLine(lineToInsert));
-
-    // Try find the lines we just deleted
-    const line = state.draft.lines.find(({ id }) => lineToInsert.id === id);
+  it('a line which is created, then deleted, is removed from state completely', () => {
+    const lineToDelete = createLine('99');
+    const state1 = callReducer(OutboundAction.upsertLine(lineToDelete));
+    const state2 = callReducer(OutboundAction.deleteLine(lineToDelete), state1);
 
-    expect(line).toEqual(
-      expect.objectContaining({
-        isCreated: true,
-        isUpdated: true,
-        isDeleted: false,
-      })
+    // Find the line in the first state, ensuring we did create it
+    const lineCreated = state1.draft.lines.find(
+      ({ id }) => lineToDelete.id === id
+    );
+    // Should not find the line in the second state
+    const lineDeleted = state2.draft.lines.find(
+      ({ id }) => lineToDelete.id === id
     );
-  });
 
-  it('adds flags to an already existing line to be updated, but not created or deleted', () => {
-    const lineToUpdate = createLine('1');
-    const state = callReducer(OutboundAction.upsertLine(lineToUpdate));
+    expect(lineCreated).toBeTruthy();
+    expect(lineDeleted).toBeFalsy();
+  });
 
-    // Try find the line we just updated
-    const line = state.draft.lines.find(({ id }) => lineToUpdate.id === id);
+  it('inserting a line for an item which has an existing, persisted, but deleted, line, reuses that existing line', () => {
+    // Mock an already existing and persisted line, which has been deleted client side.
+    const existingLine = createLine('999', {
+      itemId: 'item1',
+      numberOfPacks: 999,
+      isCreated: false,
+      isDeleted: true,
+    });
 
-    expect(line).toEqual(
-      expect.objectContaining({
-        isCreated: false,
-        isUpdated: true,
-        isDeleted: false,
-      })
+    // Simulate the user adding a new line for the same item.
+    const lineToCreate2 = createLine('998', {
+      itemId: 'item1',
+      numberOfPacks: 999,
+    });
+    const state = callReducer(
+      OutboundAction.upsertLine(lineToCreate2),
+      // Note the use of the existing line in state.
+      getState({ defaultLines: [existingLine] })
+    );
+    const lineCreated2 = state.draft.lines.find(
+      // The line should exist in the draft with the same ID and have the isDeleted tag set to false AND the isCreated tag.
+      // Note that we are matching on the original line id, not the new line id.
+      ({ id, isDeleted, isCreated }) =>
+        existingLine?.id === id && !isDeleted && isCreated
     );
```

Comment:
this is the case where the user has added a line, saved, then deleted it (not saved) then added it again 🙄  in which case we just want to send an `update`, so we reuse the line.

I think that covers all the cases unless you can think of any